adapters/huaweiads/huaweiads.go

```diff
@@ -482,9 +547,48 @@ func getHuaweiAdsReqAppInfo(request *huaweiAdsRequest, openRTBRequest *openrtb2.
 			app.Lang = "en"
 		}
 	}
-	app.Country = getCountryCode(openRTBRequest)
+	countryCode = getCountryCode(openRTBRequest)
+	app.Country = countryCode
 	request.App = app
-	return nil
+	return countryCode, nil
+}
+
+// when it has pkgNameConvert (include different rules)
+// 1. when bundleName in ExceptionPkgNames, finalPkgname = bundleName
+// 2. when bundleName conform UnconvertedPkgNames, finalPkgname = ConvertedPkgName
+// 3. when bundleName conform keyword, finalPkgname = ConvertedPkgName
+// 4. when bundleName conform prefix, finalPkgname = ConvertedPkgName
+func getFinalPkgName(bundleName string, extraInfo ExtraInfo) string {
+	for _, convert := range extraInfo.PkgNameConvert {
+		if convert.ConvertedPkgName == "" {
```

Comment:
Is there a use case for an `PkgNameConvert` entry to not have a value in `ConvertedPkgName`? If not you may want to check for that when building your adapter and throw an error if an entry is missing this field.

Reply:
thank you, parameter validation has been added, as well as test code

Reply:
yes, I know, the check code was added in the Builder() -> getExtraInfo(), getExtraInfo will return the error, Builder()  can catch, you can check.